sync/syncer.go

```diff
@@ -416,7 +416,37 @@ func (s *Syncer) validateAndBuildTx(x *types.SerializableSignedTransaction) (*ty
 
 //returns txs out of txids that are not in the local database
 func (s *Syncer) syncTxs(txids []types.TransactionId) ([]*types.AddressableSignedTransaction, error) {
+	unprocessedTxs := make(map[types.TransactionId]*types.AddressableSignedTransaction)
+
+	for out := range s.fetchWithFactory(NewNeighborhoodWorker(s, 1, TxReqFactory(txids, s))) {
+		if ntxs, ok := out.([]types.SerializableSignedTransaction); ok {
+			for _, tmp := range ntxs {
+				tx := tmp
+				ast, err := s.validateAndBuildTx(&tx)
+				if err != nil {
+					id := types.GetTransactionId(&tx)
+					s.Warning("tx %v not valid %v", hex.EncodeToString(id[:]), err)
+					continue
```

Comment:
We found out that the block contains invalid tx, we should return err and fail the block syntactic validation.